cmd/notification.go

```diff
@@ -380,7 +380,10 @@ func (sys *NotificationSys) AddRulesMap(bucketName string, rulesMap event.RulesM
 		rulesMap.Add(targetRulesMap)
 	}
 
-	rulesMap.Add(sys.bucketRulesMap[bucketName])
+	// Do not add for an empty rulesMap.
+	if len(rulesMap) > 0 {
+		rulesMap.Add(sys.bucketRulesMap[bucketName])
+	}
```

Comment:
Basically you would need to `delete()` entry in `sys.bucketRulesMap`  if `rulesMap` is empty else add it.

Reply:
ok @balamurugana  . Did the changes